src/PostgREST/Error.hs

```diff
@@ -94,180 +94,199 @@ instance PgrstError ApiRequestError where
+    mempty
+
+  toJSON PutMatchingPkError = toJsonPgrstError
+    ApiRequestErrorCode15
+    "Payload values do not match URL in primary key column(s)"
+    mempty
+    mempty
+
+  toJSON (SingularityError n) = toJsonPgrstError
+    ApiRequestErrorCode16
+    "JSON object requested, multiple (or no) rows returned"
+    (Just $ T.unwords ["The result contains", show n, "rows"])
+    mempty
+
+  toJSON (UnsupportedMethod method) = toJsonPgrstError
+    ApiRequestErrorCode17
+    ("Unsupported HTTP method: " <> T.decodeUtf8 method)
+    mempty
+    mempty
+
+  toJSON (RelatedOrderNotToOne origin target) = toJsonPgrstError
+    ApiRequestErrorCode18
+    ("A related order on '" <> target <> "' is not possible")
+    (Just $ "'" <> origin <> "' and '" <> target <> "' do not form a many-to-one or one-to-one relationship")
+    mempty
+
+  toJSON (SpreadNotToOne origin target) = toJsonPgrstError
+    ApiRequestErrorCode19
+    ("A spread operation on '" <> target <> "' is not possible")
+    (Just $ "'" <> origin <> "' and '" <> target <> "' do not form a many-to-one or one-to-one relationship")
+    mempty
+
+  toJSON (UnacceptableFilter target) = toJsonPgrstError
+    ApiRequestErrorCode20
+    ("Bad operator on the '" <> target <> "' embedded resource")
+    (Just "Only is null or not is null filters are allowed on embedded resources")
+    mempty
+
+  toJSON PGRSTParseError = toJsonPgrstError
+    ApiRequestErrorCode21
+    "The message and detail field of RAISE 'PGRST' error expects JSON"
+    mempty
+    mempty
+
+  toJSON (InvalidPreferences prefs) = toJsonPgrstError
+    ApiRequestErrorCode22
+    "Invalid preferences given with handling=strict"
+    (Just $ T.decodeUtf8 ("Invalid preferences: " <> BS.intercalate ", " prefs))
+    mempty
+
+  toJSON (NoRelBetween parent child embedHint schema allRels) = toJsonPgrstError
+    SchemaCacheErrorCode00
+    ("Could not find a relationship between '" <> parent <> "' and '" <> child <> "' in the schema cache")
+    (Just $ "Searched for a foreign key relationship between '" <> parent <> "' and '" <> child <> maybe mempty ("' using the hint '" <>) embedHint <> "' in the schema '" <> schema <> "', but no matches were found.")
+    (noRelBetweenHint parent child schema allRels)
 
   toJSON (AmbiguousRelBetween parent child rels) = JSON.object [
     "code"    .= SchemaCacheErrorCode01,
-    "message" .= ("Could not embed because more than one relationship was found for '" <> parent <> "' and '" <> child <> "'" :: Text),
+    "message" .= ("Could not embed because more than one relationship was found for '" <> parent <> "' and '" <> child <> "'"),
     "details" .= (compressedRel <$> rels),
-    "hint"    .= ("Try changing '" <> child <> "' to one of the following: " <> relHint rels <> ". Find the desired relationship in the 'details' key." :: Text)]
+    "hint"    .= ("Try changing '" <> child <> "' to one of the following: " <> relHint rels <> ". Find the desired relationship in the 'details' key.")]
+
```

Comment:
Not sure how to handle this. It could be left as-is for that case only. The issue is that  `compressedRel <$> rels` returns `[JSON.Value]` instead of `JSON.Value` or `Text` (it's the array of relationships).